src/super_gradients/training/utils/media/image.py

```diff
@@ -77,12 +83,17 @@ def load_image(image: ImageSource) -> np.ndarray:
         - str:              A string representing either a local file path or a URL to an image
 
     :param image: Single image of supported types.
-    :return:      Image as numpy arrays. If loaded from string, the image will be returned as RGB.
+    :return:      Image as numpy arrays (H, W, C). If loaded from string, the image will be returned as RGB.
     """
     if isinstance(image, np.ndarray):
+        if image.shape[0] == 3:
+            image = image.transpose((1, 2, 0))
         return image
     elif isinstance(image, torch.Tensor):
-        return image.numpy()
+        image = image.cpu().numpy()
```

Comment:
What if image is already on cuda? That would hurt the performance quite a lot. Yes the predict() was not meant for high-speed inference, but I also think we should not make it intentionally slower.

Reply:
Second comment regarding torch tensors as input - let think for a second when and why this image type may come in.
To me, the `torch.from_numpy(cv2.imread(...))` is the least common scenario. I would guess the most common use case would be an attempt to run predict on the sample from a dataset:

```
dataset = ..... # User got some dataset
sample = dataset[0]
image = sample[0]

model.predict(image)
```

And here lies the problem, as most likely a normalization was applied to this image. 
There is only one indirect clue that indicates whether or not image was normalized, which is dtype of image tensor. However I'm not sure we should be checking this.

In general, I have a strong concern how exactly we should support torch tensor types to avoid confusion.

Reply:
I agree, but we run the processings on numpy, so we don't really have a choice.

Reply:
> Second comment regarding torch tensors as input - let think for a second when and why this image type may come in.
> To me, the torch.from_numpy(cv2.imread(...)) is the least common scenario. I would guess the most common use case would be an attempt to run predict on the sample from a dataset:

That's the paradox of the `predict`; it was built with the idea of a quickstart (not really optimized) function to get a taste of YoloNAS, when in practice it is often seen as THE way to run predictions (and benchmark) on super-gradients.
So, supporting dataset output was never the target but I agree that some users may want or even already try to do so.

My approach to this would be: lets keep as it is now until we get enough feedback from people wanting/complaining about that, and then we can discuss about it more in-depth